RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryAlgos.h

```diff
@@ -7,16 +7,27 @@
 #include "DataFormats/EcalDetId/interface/EBDetId.h"
 #include <string>
 
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
 #include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryNN.h"
+#include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryBDTG.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 template <typename DetIdT>
 class EcalDeadChannelRecoveryAlgos {
 public:
+  void setParameters(const edm::ParameterSet &ps);
   void setCaloTopology(const CaloTopology *topology);
-  EcalRecHit correct(
-      const DetIdT id, const EcalRecHitCollection &hit_collection, std::string algo, double Sum8Cut, bool *AccFlag);
+  EcalRecHit correct(const DetIdT id,
+                     const EcalRecHitCollection &hit_collection,
+                     std::string algo,
+                     double single8Cut,
+                     double sum8Cut,
+                     bool *accFlag);
 
 private:
-  EcalDeadChannelRecoveryNN<DetIdT> nn;
+  EcalDeadChannelRecoveryNN<DetIdT> nn_;
+  EcalDeadChannelRecoveryBDTG<DetIdT> bdtg_;
```

Comment:
I do not quite understand why both EcalDeadChannelRecoveryNN and EcalDeadChannelRecoveryBDTG have to be loaded here.
The configuration implies that only one method is available.
IIUC, EcalDeadChannelRecoveryNN carries some significant cost in initialization already even if it is not used.
It looks more practical to change these (NN and BDTG) to unique_ptr and initialize only the one that's actually used.